Vowpal Wabbit - Contextual Bandits.ipynb

```diff
@@ -0,0 +1,705 @@
+      "weighted example sum = 0.000000\n",
+      "weighted label sum = 0.000000\n",
+      "average loss = n.a.\n",
+      "total feature number = 0\n",
+      "23/01/16 13:53:10 WARN VowpalWabbitGeneric: VowpalWabbit args: --cb_adf --cb_type mtr --clip_p 0.1 -q GT -q MS -q GR -q OT -q MT -q OS --dsjson --preserve_performance_counters --no_stdin)\n",
+      "creating quadratic features for pairs: GT MS GR OT MT OS            (0 + 1) / 1]\n",
+      "creating quadratic features for pairs: GT MS GR OT MT OS\n",
+      "using no cache\n",
+      "Reading datafile = none\n",
+      "num sources = 0\n",
+      "Num weight bits = 18\n",
+      "learning rate = 0.5\n",
+      "initial_t = 0\n",
+      "power_t = 0.5\n",
+      "cb_type = mtr\n",
+      "Enabled reductions: gd, scorer-identity, csoaa_ldf-rank, cb_adf, shared_feature_merger\n",
+      "Input label = cb\n",
+      "Output pred = action_scores\n",
+      "average  since         example        example        current        current  current\n",
+      "loss     last          counter         weight          label        predict features\n",
+      "0.000000 0.000000            1            1.0       9:0:0.82            0:0     1791\n",
+      "using no cache\n",
+      "Reading datafile = none\n",
+      "num sources = 0\n",
+      "Num weight bits = 18\n",
+      "learning rate = 0.5\n",
+      "initial_t = 0\n",
+      "power_t = 0.5\n",
+      "cb_type = mtr\n",
+      "Enabled reductions: gd, scorer-identity, csoaa_ldf-rank, cb_adf, shared_feature_merger\n",
+      "Input label = cb\n",
+      "Output pred = action_scores\n",
+      "average  since         example        example        current        current  current\n",
+      "loss     last          counter         weight          label        predict features\n",
+      "0.000000 0.000000            1            1.0      9:-1:0.82            0:0     1791\n",
+      "\n",
+      "finished run\n",
+      "number of examples = 1\n",
+      "weighted example sum = 1.000000\n",
+      "weighted label sum = 0.000000\n",
+      "average loss = 0.000000\n",
+      "total feature number = 1726\n",
+      "0.000000 0.000000            2            2.0      9:-1:0.82        6:-2.67     1791\n",
+      "\n",
+      "finished run\n",
+      "number of examples = 2\n",
+      "weighted example sum = 2.000000\n",
+      "weighted label sum = 0.000000\n",
+      "average loss = 0.000000\n",
+      "total feature number = 3452\n",
+      "23/01/16 13:53:11 WARN VowpalWabbitGeneric: VowpalWabbit args: --cb_adf --cb_type mtr --clip_p 0.1 -q GT -q MS -q GR -q OT -q MT -q OS --dsjson --preserve_performance_counters --quiet --no_stdin)\n"
+     ]
+    }
+   ],
+   "source": [
+    "from synapse.ml.vw import VowpalWabbitGeneric\n",
+    "\n",
+    "model = (\n",
+    "    VowpalWabbitGeneric(\n",
+    "        passThroughArgs=\"--cb_adf --cb_type mtr --clip_p 0.1 -q GT -q MS -q GR -q OT -q MT -q OS --dsjson --preserve_performance_counters\"\n",
```

Comment:
are there any args that can be set without pass-through? Perhaps also use the .setPassThroughArgs here for consistency.

Reply:
a) these params are not exposed through setters.
b) VW user will copy&paste the command line from other envs

in general I think the setters are only useful in hyper parameter search settings.